src/components/Table/DrawerListTable/index.vue

```diff
@@ -233,29 +233,38 @@ export default {
     },
     getRouteNameComponent(name, action) {
       const route = { name: name }
+
       if (action === 'detail' || action === 'update') {
         route.params = { id: '1' }
       }
-      return resolveRoute(route, this.$router)
+
+      const resolved = resolveRoute(route, this.$router)
+
+      if (resolved && resolved.components && resolved.components.default) {
+        return resolved.components.default
+      }
+
+      return ''
     },
     getDetailComponent({ detailRoute }) {
       if (!detailRoute) {
         return this.detailDrawer
       }
-      this.$log.debug('>>> getDetailComponent: ', detailRoute)
+
       const route = resolveRoute(detailRoute, this.$router)
       let component = null
-      if (route) {
+
+      if (route && route.components && route.components.default) {
         component = route.components.default
       }
+
       if (!component) {
         component = this.detailDrawer
       }
+
       return component
     },
     getDrawerComponent(action, payload) {
-      this.$log.debug('>>> getDrawerComponent: ', action, payload)
-      // console.log('>>> createDrawer: ', this.createDrawer)
       switch (action) {
         case 'create':
           return this.createDrawer
```

Comment:
It looks like there is no obvious issue or irregularity with the provided code snippet. However, it's best to run some benchmarks before making significant changes to verify performance improvements.

For optimization suggestions, you could consider implementing memoization for function calls if they are called multiple times in a row.

Overall, no major optimizations seem necessary. The current structure and implementation should be efficient based on my knowledge cut-off point of September 1st, 2021.